src/main/java/de/tum/in/www1/artemis/service/LectureService.java

```diff
@@ -133,6 +140,14 @@ public SearchResultPageDTO<Lecture> getAllOnPageWithSize(final SearchTermPageabl
      * @param lecture the lecture to be deleted
      */
     public void delete(Lecture lecture) {
+        if (pyrisWebhookService.isPresent()) {
+            Lecture lectureWithAttachmentUnits = lectureRepository.findByIdWithLectureUnitsAndAttachmentsElseThrow(lecture.getId());
+            List<AttachmentUnit> attachmentUnitList = lectureWithAttachmentUnits.getLectureUnits().stream().filter(lectureUnit -> lectureUnit.getType().equals("attachment"))
+                    .map(lectureUnit -> (AttachmentUnit) lectureUnit).collect(Collectors.toCollection(ArrayList::new));
+            if (!attachmentUnitList.isEmpty()) {
+                pyrisWebhookService.get().executeIngestionPipeline(false, attachmentUnitList);
+            }
+        }
```

Comment:
For anyone not familiar with Iris it would help to add a comment what this does.
Also makes sense for other calls to `pyrisWebhookService `